schemas/stsci.edu/asdf/0.1.0/core/table.yaml

```diff
@@ -0,0 +1,108 @@
+%YAML 1.1
+---
+$schema: "http://stsci.edu/schemas/yaml-schema/draft-01"
+id: "http://stsci.edu/schemas/asdf/0.1.0/core/table"
+tag: "tag:stsci.edu:asdf/0.1.0/core/table"
+
+title: >
+  A table.
+
+description: |
+  A table is represented as a list of columns, where each entry is a
+  [column](ref:http://stsci.edu/schemas/asdf/0.1.0/core/column)
+  object, containing the data and some additional information.
+
+  The data itself may be stored inline as text, or in binary in either
+  row- or column-major order by use of the `strides` property on the
+  individual column arrays.
+
+  Each column in the table must have the same first (slowest moving)
+  dimension.
+
+examples:
+  -
+    - A table stored in column-major order, with each column in a separate block
+    - |
+        !core/table
+          columns:
+          - !core/column
+            data: !core/ndarray
+              source: 0
+              datatype: float64
+              byteorder: little
+              shape: [3]
+            description: RA
+            meta: {foo: bar}
+            name: a
+            unit: !unit/unit deg
+          - !core/column
+            data: !core/ndarray
+              source: 1
+              datatype: float64
+              byteorder: little
+              shape: [3]
+            description: DEC
+            name: b
+          - !core/column
+            data: !core/ndarray
+              source: 2
+              datatype: [ascii, 1]
+              byteorder: big
+              shape: [3]
+            description: The target name
+            name: c
+
+  -
+    - A table stored in row-major order, all stored in the same block
+    - |
+        !core/table
+          columns:
+          - !core/column
+            data: !core/ndarray
+              source: 0
+              datatype: float64
+              byteorder: little
+              shape: [3]
+              strides: [13]
+            description: RA
+            meta: {foo: bar}
+            name: a
+            unit: !unit/unit deg
+          - !core/column
+            data: !core/ndarray
+              source: 0
+              datatype: float64
+              byteorder: little
+              shape: [3]
+              offset: 4
+              strides: [13]
+            description: DEC
+            name: b
+          - !core/column
+            data: !core/ndarray
+              source: 0
+              datatype: [ascii, 1]
+              byteorder: big
+              shape: [3]
+              offset: 12
+              strides: [13]
+            description: The target name
+            name: c
+
+type: object
+properties:
+  columns:
+    description: |
+      A list of columns in the table.
+    type: array
+    items:
+      $ref: column
+
+  meta:
+    description: |
+      Additional free-form metadata about the table.
+    type: object
+    default: {}
```

Comment:
So now that I play around with this, I'm thinking maybe that something else the table schema needs to allow is a `data` property.  That way we can write all the data for a table in one place as an inline array that the individual columns can reference.

Currently if you write a table with inline, each column's data is written inline with the column, but we lose the nice inline row-based table.  I think an argument could be made for both representations, but I think probably keeping the row-based output will be more common.  

One thing that isn't immediately obvious to me is how to tell the writer which way to write the table.  I fear an additional write option might be necessary for that :/